canlink/busmanager.go

```diff
@@ -0,0 +1,242 @@
+// Package canlink provides utilities to interact with a
+// Controller Area Network (CAN Bus).
+package canlink
+
+import (
+	"context"
+	"errors"
+	"net"
+	"sync"
+	"time"
+
+	"go.einride.tech/can/pkg/socketcan"
+	"go.uber.org/zap"
+)
+
+// Default buffered channel length for the broadcast and
+// incoming receive channels.
+//
+// In order to preserve the real time broadcast requirement,
+// BusManager will drop messages for any Handler that is
+// unable to keep up with the broadcast rate.
+const _channelBufferLength = 1000
+
+// BusManager is a centralized node responsible for orchestrating
+// all interactions with a CAN bus.
+//
+// It acts as a message broker supporting the transmission
+// of bus traffic to registered handlers and receiving incoming
+// messages from these handlers to write to the bus.
+//
+// BusManager uses SocketCAN on the Linux platform. Note that
+// it does not manage the lifetime of the network socket connection.
+//
+// Example:
+//
+//	 package main
+//
+//	 import (
+//	   "context"
+//	   "fmt"
+//	   "time"
+//
+//	   "go.einride.tech/can/pkg/socketcan"
+//	   "go.uber.org/zap"
+//	 )
+//
+//	 func main () {
+//	   ctx := context.Background()
+//
+//	   loggerConfig := zap.NewDevelopmentConfig()
+//	   logger, err := loggerConfig.Build()
+//
+//		 // Create a network connection for vcan0
+//		 conn, err := socketcan.DialContext(context.Background(), "can", "vcan0")
+//		 if err != nil {
+//		   return
+//		 }
+//
+//	   manager := canlink.NewBusManager(logger, conn)
+//	   handler = NewHandler(...)
+//
+//	   broadcast, transmit := manager.Register(handler)
```

Comment:
shouldn't this be something like 
```suggestion
//	  receiveChan, transmitChan := manager.Register(handler)
```

Reply:
Agreed, will update.